orchagent/pfcwdorch.cpp

```diff
@@ -39,6 +39,42 @@ PfcWdOrch<DropHandler, ForwardHandler>::PfcWdOrch(DBConnector *db, vector<string
     timer->start();
 }
 
+template <typename DropHandler, typename ForwardHandler>
+QueueMcDropCounters PfcWdOrch<DropHandler, ForwardHandler>::getQueueMcDropCounters(
+        const Port& port,
+        uint8_t pfcMask)
+{
+    SWSS_LOG_ENTER();
+
+    vector<FieldValueTuple> fieldValues;
+    QueueMcDropCounters counters;
+    counters.fill(numeric_limits<uint64_t>::max());
+
+    for (uint8_t prio = 0; prio < PFC_WD_TC_MAX; prio++)
+    {
+        sai_object_id_t queueId = port.m_queue_ids[prio + PFC_WD_TC_MAX];
```

Comment:
In broadcom platform, there are 10 uc queues and 10 mc queues. Total queue number is not always equal to PFC_WD_TC_MAX. I am not sure it's the correct way to get queueId here. Maybe consider to get port queue list first?

Reply:
Done